core/appcontainer/OptionsDialogModel.js

```diff
@@ -0,0 +1,152 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2018 Extremely Heavy Industries Inc.
+ */
+import {XH, HoistModel} from '@xh/hoist/core';
+import {FormModel} from '@xh/hoist/cmp/form';
+import {observable, computed, action} from '@xh/hoist/mobx';
+import {isFunction} from 'lodash';
+
+import {AppOption} from './AppOption';
+
+/**
+ * Manages built-in setting of user preferences.
+ * @private
+ */
+@HoistModel
+export class OptionsDialogModel {
+
+    @observable isOpen = false;
+    @observable.ref options = [];
+    @observable.ref changes = [];
+
+    formModel = new FormModel();
+
+    //-------------------
+    // Setting options
+    //-------------------
+    @action
+    setOptions(options) {
+        // Ensure each is valid AppOption
+        this.options = options.map(it => it instanceof AppOption ? it : new AppOption(it));
+
+        // Add each AppOption to the FormModel
+        this.options.forEach(it => this.formModel.addField(it));
+
+        // Add change detection reaction
+        this.addReaction({
+            track: () => this.formModel.getData(),
+            run: () => this.refreshChangedFields()
+        });
+    }
+
+    @computed
+    get hasOptions() {
+        return !!this.options.length;
+    }
+
+    //-------------------
+    // Change management
+    //-------------------
+    @action
+    refreshChangedFields() {
```

Comment:
I don't get the need for the extra state in the changes array.  Couldn't we just walk through the fields, both for the computed "requiresRefresh" and saveAsync()?